### ISSUE TITLE
Rename via the new renamer API should dismiss renaming tracking

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#44469]
After renaming a type to match the class name, the type gets treated as if it was manually renamed and enters a confusing behavior where I get a code fix to perform the rename that I just performed.

**Steps to Reproduce**:

1. Checkout and build https://github.com/dotnet/project-system/pull/6179 
2. Create new Console App (.NET Core) and rename Program.cs -> Bar.cs and say Yes to the prompt

**Expected Behavior**:
![image](https://user-images.githubusercontent.com/1103906/82518320-775e7200-9b62-11ea-9313-f6b1e9ff9697.png)

**Actual Behavior**:
![image](https://user-images.githubusercontent.com/1103906/82518307-71689100-9b62-11ea-8c09-2149531e334c.png)

Legacy does not run into this because it uses CodeModel to rename which dismisses the tracking here: http://sourceroslyn.io/#Microsoft.VisualStudio.LanguageServices.Implementation/CodeModel/AbstractCodeModelService.cs,505.

//cc @ryzngard @ocallesp

**Comment by ryzngard** [comment]
Thanks! Looking

**Comment by CyrusNajmabadi** [comment]
@ryzngard still needed?

**Comment by ryzngard** [comment]
Yes, but it's low pri work right now. User experience is a bug, but manageable and low impact.
